src/lib/y2network/dialogs/additional_address.rb

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "cwm/popup"
+require "y2network/widgets/ip_address"
+
+module Y2Network
+  module Dialogs
+    # Popup dialog to add or edit an additional IP address configuration to a
+    # connection config
+    class AdditionalAddress < CWM::Popup
+      # Constructor
+      #
+      # @param name [String]
+      # @param settings [Object]
+      def initialize(name, settings)
+        textdomain "network"
+
+        @name     = name
+        @settings = settings
+      end
+
+      def contents
+        VBox(
+          label_widget,
+          ip_address_widget,
+          netmask_widget
+        )
+      end
+
+      def buttons
+        [ok_button, cancel_button]
+      end
+
+      def label_widget
+        @label_widget ||= IPAddressLabel.new(@name, @settings)
+      end
+
+      def ip_address_widget
+        @ip_address_widget ||= IPAddress.new(@settings)
+      end
+
+      def netmask_widget
+        @netmask_widget ||= Y2Network::Widgets::Netmask.new(@settings)
+      end
+    end
+
+    class IPAddress < Y2Network::Widgets::IPAddress
+      def init
+        super
+        focus unless @settings.label.to_s.empty?
+      end
+    end
+
+    # Widget to modify the label of an additional IP address configuration
+    class IPAddressLabel < CWM::InputField
+      def initialize(name, settings)
+        textdomain "network"
+
+        @name = name
+        @settings = settings
+      end
+
+      def init
+        self.value = @settings.label
+        focus if @settings.label.to_s.empty?
```

Comment:
to be honest I do not like much that focus is handled on this level. What about having focus as constructor parameter and just focus in init when requested in constructor? ( if it is not used top level but in e.g. tab, then it can be handled on that level )